src/main/kotlin/org/pkl/lsp/packages/dto/PackageAssetUri.kt

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.lsp.packages.dto
+
+import java.net.URI
+import java.net.URISyntaxException
+
+data class PackageAssetUri(val packageUri: PackageUri, val assetPath: String) {
+  companion object {
+    fun create(str: String): PackageAssetUri? {
+      val uri =
+        try {
+          URI(str)
+        } catch (e: URISyntaxException) {
+          return null
+        }
```

Comment:
Do you know about `runCatching`?
```suggestion
      val uri = runCatching { URI(str) }.getOrNull() ?: return null
```